src/Components/Server/src/ComponentHub.cs

```diff
@@ -143,7 +166,17 @@ private async void CircuitHost_UnhandledException(object sender, UnhandledExcept
             try
             {
                 Log.UnhandledExceptionInCircuit(_logger, circuitId, (Exception)e.ExceptionObject);
-                await circuitHost.Client.SendAsync("JS.Error", e.ExceptionObject);
+                if (_options.DetailedErrors)
+                {
+                    await circuitHost.Client.SendAsync("JS.Error", e.ExceptionObject.ToString());
+                }
+                else
+                {
+                    var message = $"There was an error on the current session. For more details turn on " +
```

Comment:
```suggestion
                    var message = $"There was an unhandled exception on the current circuit, so this circuit will be terminated. For more details turn on " +
```

Reply:
Where do we terminate the circuit? That doesn't happen AFAIK, does it?

Reply:
It's https://github.com/aspnet/AspNetCore/issues/11845